### PR TITLE
Add complete Game

### PR DESCRIPTION
**Pull request by MarcRaeder**
Add complete Game

### CODE REVIEW

File: zahlenRaten.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    elif distanceGuessToNumber < distanceLastGuessToNumber:
+        print("wärmer")
+
+
+def game():
+
+    number = random.randint(1, 100)
+    trys = 0
+    min = 1
+    max = 100
```
Comment (DevBastiS) [inline zahlenRaten.py]: Können als Konstanten oben in der Datei deklariert werden und einen sprechenderen Namen bekommen.

File: zahlenRaten.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import random
+
+
+def compareDistance(distanceGuessToNumber, distanceLastGuessToNumber, number):
+
```
Comment (DevBastiS) [inline zahlenRaten.py:8]:
```suggestion
```

File: zahlenRaten.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    if distanceGuessToNumber > distanceLastGuessToNumber:
+        print("kälter")
+    elif distanceGuessToNumber < distanceLastGuessToNumber:
+        print("wärmer")
+
+
+def game():
+
+    number = random.randint(1, 100)
+    trys = 0
```
Comment (DevBastiS) [inline zahlenRaten.py]:
```suggestion
    tries = 0
```

oder "versuche" oder "attempts"

**Review by DevBastiS — CHANGES_REQUESTED** [review]

File: zahlenRaten.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+                decision = str(input())
+
+                if decision.upper() == "JA":
+                    trys = 0
+                    game()
+                else:
+                    print("Auf Wiedersehen")
+                    break
+                break
```
Comment (stefanluth) [inline zahlenRaten.py]: Calling `game()` inside `game()` will cause nested function instances. This means when you play 20 rounds in a row, you will have 20 running games, all stuck at line 71 and then quitting would quit all 20 games at once. It's preferable to only have one instance of the game at any given time.

**Review by stefanluth — CHANGES_REQUESTED** [review]

File: zahlenRaten.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+            elif currentGuess > number:
+                print("Deine Zahl ist zu hoch. Versuch es erneut!")
+                tries += 1
+                distanceLastGuessToNumber = abs(lastGuess - number)
+                compareDistance(distanceGuessToNumber,
+                                distanceLastGuessToNumber, number)
+                lastGuess = currentGuess
+            elif currentGuess == number:
+                print("Das war richtig. Nice! Du hast",
+                      tries, "Versuche gebraucht.\nMöchtest du nochmal spielen? Dann gebe 'JA' ein.\nFalls nicht dann gib irgendwas anderes ein.")
```
Comment (stefanluth) [inline zahlenRaten.py]:
```suggestion
                print("Das war richtig. Nice! Du hast",
                    tries, "Versuche gebraucht.\nMöchtest du nochmal spielen? Dann gebe 'JA' ein.\nFalls nicht dann gib irgendwas anderes ein.")
```
Indentation is off and the line is kinda long :)

File: zahlenRaten.py [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+game()
+
+decision = str(input())
+
+if decision.upper() == "JA":
+    tries = 0
+    game()
+else:
+    print("Auf Wiedersehen")
```
Comment (stefanluth) [inline zahlenRaten.py]: The game will only be playable one more time, instead of having an endless loop

**Review by stefanluth — APPROVED** [review]

**Review by DevBastiS — APPROVED** [review]

**Pull request merged**